projects/igniteui-angular/src/lib/services/direction/directionality.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Inject, InjectionToken, inject } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+
+/**
+ * @hidden
+ */
+export type Direction = 'ltr' | 'rtl';
+
+/**
+ * Injection token is used to inject the document into Directionality
+ * which factory could be faked for testing purposes.
+ *
+ * We can't provide and mock the DOCUMENT token from platform-browser because configureTestingModule
+ * allows override of the default providers, directive, pipes, modules of the test injector
+ * which causes errors.
+ *
+ * @hidden
+ */
+export const DIR_DOCUMENT = new InjectionToken<Document>('dir-doc', {
+    providedIn: 'root',
+    factory: DIR_DOCUMENT_FACTORY
+});
+
+/**
+ * @hidden
+ */
+export function DIR_DOCUMENT_FACTORY(): Document {
+    return inject(DOCUMENT);
+}
+
+/**
+ * @hidden
+ * @dynamic
```

Comment:
If you worked around the metadata error with a type cast, you don't really need `@dynamic`, right?
https://github.com/angular/angular/issues/26874 Seems to cover just this error but in case it affects others in the future - better remove it.